R/add_global_p.R

```diff
@@ -310,3 +245,57 @@ add_global_p.tbl_uvregression <- function(x, type = NULL, include = everything()
   x$call_list <- updated_call_list
   x
 }
+
+tidy_car_anova <- function(x, type, tbl, ...) {
+  car::Anova(mod = x, type = type, ...) %>%
+    {suppressWarnings(broom::tidy(.))}
+}
+
+# this function runs anova_fun if specified.
+# if anova_fun is NULL, then we use car::Anova,
+.run_anova <- function(x, type, anova_fun, variable = NULL, ...) {
+  dots <- rlang::dots_list(...)
+
+  # running custom function passed by user
+  if (!is.null(anova_fun)) {
+    result <-
+      tryCatch(
+        do.call(anova_fun, args = c(list(x = x), dots)),
+        error = function(e) {
+          ifelse(
+            !is.null(variable),
+            "There was an error running {.code anova_fun()} for {.val {variable}}",
+            "There was an error running {.code anova_fun()}"
+          ) %>%
+          cli::cli_alert_danger()
+          stop(e)
+        }
+      )
+    return(result)
+  }
+
+  # running car::Anova()
+  tryCatch(
+    do.call(car::Anova, args = c(list(mod = x, type = type), dots)) %>% broom::tidy(),
+
+    # trying `add_global_p(anova_fun = gtsummary::tidy_wald_test)` if `car::Anova()` fails
+    error = function(e) {
+      ifelse(
+        !is.null(variable),
+        "There was an error running {.code car::Anova()} for {.val {variable}}",
+        "There was an error running {.code car::Anova()}"
+      ) %>%
+        paste0(", trying {.code add_global_p(anova_fun = gtsummary::tidy_wald_test)}") %>%
```

Comment:
What do you think of updating the messaging from "There was an error running `car::Anova()`, trying `add_global_p(anova_fun = gtsummary::tidy_wald_test)`" to something like one of the following

* "There was an error running `car::Anova()`. The results displayed are based on `add_global_p(anova_fun = gtsummary::tidy_wald_test)`" 
* "There was an error running `car::Anova()`, likely due to this model type not being supported. The results displayed are based on `add_global_p(anova_fun = gtsummary::tidy_wald_test)`"?

Reply:
@jalavery sounds good! updated wording has been pushed